### PR TITLE
Documenting Spans and Transactions

### PR DESCRIPTION
**Pull request by antonpirker**
Added some docstrings and also did some minor cleanup for better readability of the code.

### CODE REVIEW

File: sentry_sdk/hub.py [hunk 1]
```diff
@@ -487,24 +488,33 @@ def start_span(self, span=None, instrumenter=INSTRUMENTER.SENTRY, **kwargs):
...
+
             if "transaction" in kwargs:
                 logger.warning(deprecation_msg)
                 name = kwargs.pop("transaction")
                 return self.start_transaction(name=name, **kwargs)
 
+        # THIS BLOCK IS DEPRECATED
+        # We do not pass a span into start_span in our code base, so I deprecate this.
         if span is not None:
+            deprecation_msg = "Deprecated: passing a span into `start_span` is deprecated and will be removed in the future. "
```
Comment (sentrivana) [inline sentry_sdk/hub.py:504]:
```suggestion
            deprecation_msg = "Deprecated: passing a span into `start_span` is deprecated and will be removed in the future."
```

File: sentry_sdk/tracing.py [hunk 2]
```diff
@@ -254,12 +263,15 @@ def continue_from_environ(
...
 
-        This is different from `continue_from_headers` in that it assumes header
-        names in the form "HTTP_HEADER_NAME" - such as you would get from a wsgi
-        environ - rather than the form "header-name".
+        This is different from :py:meth:`~sentry_sdk.tracing.Span.continue_from_headers`
+        in that it assumes header names in the form ``HTTP_HEADER_NAME`` -
+        such as you would get from a WSGI/ASGI environ -
+        rather than the form ``header-name``.
+
+        :param envrion: The ASGI/WSGI environ to pull information from.
```
Comment (sentrivana) [inline sentry_sdk/tracing.py:274]:
```suggestion
        :param environ: The ASGI/WSGI environ to pull information from.
```

File: sentry_sdk/tracing.py [hunk 3]
```diff
@@ -491,6 +522,9 @@ def get_trace_context(self):
 
 
 class Transaction(Span):
+    """The Transaction is the root element that holds all the spans
+    for the Sentry performance instrumentation."""
```
Comment (sentrivana) [inline sentry_sdk/tracing.py:526]:
```suggestion
    for Sentry performance instrumentation."""
```

File: sentry_sdk/tracing.py [hunk 4]
```diff
@@ -512,6 +546,19 @@ def __init__(
...
         # type: (...) -> None
+        """Constructs a new Transaction.
+
+        :param name: Identifier of the transaction.
+            Will show up in the Sentry UI.
+        :param parent_sampled: Whether the parent transaction was sampled.
+            If True this transaction will be kept, if False it will be discarded.
+        :param baggage: The W3C baggage header value.
+            (see https://www.w3.org/TR/baggage/)
+        :param source: A string describing the source of the transactions name.
```
Comment (sentrivana) [inline sentry_sdk/tracing.py:557]:
```suggestion
        :param source: A string describing the source of the transaction name.
```

File: sentry_sdk/tracing.py [hunk 5]
```diff
@@ -568,6 +615,9 @@ def __exit__(self, ty, value, tb):
     @property
     def containing_transaction(self):
         # type: () -> Transaction
+        """The root element of the span tree.
+        In the case of the transaction, it is the transaction itself.
```
Comment (sentrivana) [inline sentry_sdk/tracing.py:619]:
```suggestion
        In the case of a transaction it is the transaction itself.
```

File: sentry_sdk/tracing.py [hunk 6]
```diff
@@ -680,10 +752,12 @@ def to_json(self):
 
     def get_baggage(self):
         # type: () -> Baggage
-        """
+        """Returns the :py:class:`~sentry_sdk.tracing_utils.Baggage`
+        associated with the Transaction.
+
         The first time a new baggage with sentry items is made,
```
Comment (sentrivana) [inline sentry_sdk/tracing.py:758]:
```suggestion
        The first time a new baggage with Sentry items is made,
```

File: sentry_sdk/tracing_utils.py [hunk 7]
```diff
@@ -212,6 +212,10 @@ def _format_sql(cursor, sql):
 
 
 class Baggage(object):
+    """
+    The W3C Baggage header information. (see https://www.w3.org/TR/baggage/)
```
Comment (sentrivana) [inline sentry_sdk/tracing_utils.py:216]:
```suggestion
    The W3C Baggage header information (see https://www.w3.org/TR/baggage/).
```

File: sentry_sdk/tracing.py [hunk 8]
```diff
@@ -422,8 +438,21 @@ def is_success(self):
 
     def finish(self, hub=None, end_timestamp=None):
         # type: (Optional[sentry_sdk.Hub], Optional[datetime]) -> Optional[str]
-        # XXX: would be type: (Optional[sentry_sdk.Hub]) -> None, but that leads
+        # Note: would be type: (Optional[sentry_sdk.Hub]) -> None, but that leads
         # to incompatible return types for Span.finish and Transaction.finish.
+        """Sets the end timestamp of the span.
+        Additionally it also creates a breadcrumb from the span,
+        if the span represents a database or http request.
```
Comment (sentrivana) [inline sentry_sdk/tracing.py:445]:
```suggestion
        if the span represents a database or HTTP request.
```

**Review by sentrivana — APPROVED** [review]
This is so nice!

Left some typo/grammar/prettifying suggestions.

**Comment by antonpirker** [comment]
Thanks for the fixes! Yes, at some point our lib will be a really well documented thing!

**Pull request merged**